src/sessions.ts

```diff
@@ -470,10 +470,7 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    * @param options - optional settings for the transaction
    * @returns A raw command response or undefined
    */
-  withTransaction<T = void>(
-    fn: WithTransactionCallback<T>,
-    options?: TransactionOptions
-  ): Promise<Document | undefined> {
+  withTransaction<T>(fn: WithTransactionCallback<T>, options?: TransactionOptions): Promise<T> {
```

Comment:
The docs need a little update too:
```
   * @remarks
   * This function:
   * - Will return the result of the withTransaction callback if every operation is successful
   * - Will return `undefined` if the transaction is intentionally aborted with `await session.abortTransaction()`
   * - Will throw if one of the operations throws or `throw` statement is used inside the `withTransaction` callback
```

Which makes me question, should we require that the callback return something truthy (via TS) or default to something truthy (maybe `true`) if the callback returns `undefined`? Otherwise, how can someone differentiate an intentionally aborted transaction and a committed one if they don't have a return value?

[Spec ref](https://github.com/mongodb/specifications/blob/master/source/transactions-convenient-api/transactions-convenient-api.rst#the-callback-function-has-a-flexible-signature)

Reply:
I think returning `true` sounds good if they haven't returned anything.

Reply:
Sounds good to me, I'm thinking we can just add `value ?? true` and a test for that case along with docs. I'll update the ticket AC just so we keep track